internal/engine/module_builder.go

```diff
@@ -305,3 +306,43 @@ func (b *ModuleBuilder) GetContainerImages(value cue.Value) ([]string, error) {
 
 	return images, nil
 }
+
+// GetConfigStructure extracts the config structure from the module.
+func (b *ModuleBuilder) GetConfigStructure(value cue.Value) ([][]string, error) {
+	cfgValues := value.LookupPath(cue.ParsePath(apiv1.ConfigValuesSelector.String()))
+	if cfgValues.Err() != nil {
+		return nil, fmt.Errorf("lookup %s failed: %w", apiv1.ConfigValuesSelector, cfgValues.Err())
+	}
+
+	var rows [][]string
+	configDataInfo := func(v cue.Value) bool {
+		var row []string
+
+		if v.Kind().String() != "struct" && !v.IsConcrete() {
+			defaultVal, _ := v.Default()
+			valueBytes, _ := defaultVal.MarshalJSON()
+			valueType := strings.ReplaceAll(v.IncompleteKind().String(), "|", "\\|")
+			value := strings.ReplaceAll(string(valueBytes), "\":", "\": ")
+			value = strings.ReplaceAll(value, "\":[", "\": [")
+			value = strings.ReplaceAll(value, "},", "}, ")
+			value = strings.ReplaceAll(value, "|", "\\|")
+
+			row = append(row, fmt.Sprintf("`%s:`", strings.ReplaceAll(strings.Replace(v.Path().String(), "timoni.instance.config.", "", 1), ".", ": ")))
+			row = append(row, fmt.Sprintf("`%s`", valueType))
+			row = append(row, fmt.Sprintf("`%s`", value))
+
+			var doc string
+			for _, d := range v.Doc() {
+				doc += strings.ReplaceAll(strings.ReplaceAll(d.Text(), "+optional", ""), "+required", "")
+			}
+
+			row = append(row, fmt.Sprintf("%s", strings.ReplaceAll(doc, "\n", " ")))
+			rows = append(rows, row)
+		}
```

Comment:
I need to refactor this, mainly the `!v.IsConcrete()` check as this hides any required fields and the field name formatting so as not to have any labels in a domain format changed.

So a question I have for you @stefanprodan, should the timoni required fields be hidden from this table?

By timoni required I mean `kubeVersion`, `clusterVersion`, `moduleVersion` and any others I missed or may be added in the future?

Should we also hide the `metadata` group as most of this is filled in by timoni, but show `metadata: labels:` and `metadata: annotations:`?

Reply:
Looking back a the referenced issue there is mention of adding `// DOC` above comments, would the preference be to implement this and only list fields that have this as part of their comment?

It would avoid needing to build out a lot of logic on what fields to include/exclude and push that to the module maintainers.

Reply:
Let's also rename the function to `GetConfigDoc`.

Reply:
Awesome, will hopefully get something for this done today :+1:

Reply:
If we're using `+nodoc` as a convention should we also use `+optional`/`+required` and have those feed into the table? Though that would inflate the table more. My thought there being to add a new column `Required` that would be `true`/`false` based on `+optional`/`+required` being set.